cypress/plugins/index.js

```diff
@@ -178,6 +185,36 @@ function deleteTestData(currentTestRoot, app) {
       app.firestore().doc(testPrefix + currentTestRoot));
 }
 
+const millisecondsInADay = 60 * 60 * 24 * 1000;
+
+/**
+ * Recursively deletes all test data older than 24 hours, under the assumption
+ * that no test runs for that long. This prevents old unfinished tests from
+ * using too much quota. Note that documents must have a field set to show up in
+ * a listing of the parent collection
+ * (https://stackoverflow.com/questions/47043651/this-document-does-not-exist-and-will-not-appear-in-queries-or-snapshots-but-id).
+ * That field is set in clearAndPopulateTestFirestoreData.
+ * @param {admin.app.App} app
+ * @return {Promise} Promise that completes when all deletions are finished
+ */
+function deleteAllOldTestData(app) {
+  const currentDate = new Date();
+  const querySnapshotPromise = app.firestore().collection(testPrefix).get();
+  return querySnapshotPromise.then((queryResult) => {
+    const promises = [];
+    queryResult.forEach((doc) => {
+      const ref = doc.ref;
+      const testRunName = ref.id;
+      const dateElement = testRunName.split('-')[0];
+      const date = new Date(parseInt(dateElement, 10));
```

Comment:
is the 10 necessary?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt